### PR TITLE
Format environment variables to match docker-cli.

### LINKED ISSUE #948: Support environment variables without values
See https://github.com/docker/compose/pull/2997 for some context.

docker-py assumes that all variables should have an `=` but the docker-cli passes environment variables without values as just the key, which is super weird, but apparently swarm is relying on that currently.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #948

### CODE REVIEW

File: tests/integration/container_test.py [hunk 1]
```diff
@@ -368,7 +368,17 @@ def test_create_host_config_exception_raising(self):
                           self.client.create_host_config, mem_swappiness='40')
 
         self.assertRaises(ValueError,
-                          self.client.create_host_config, pid_mode='40')
+                           self.client.create_host_config, pid_mode='40')
```
Comment (shin-) [inline tests/integration/container_test.py]: You've got an extra space here making flake8 break.

**Comment by shin-** [comment]
LGTM!

File: docker/utils/utils.py [hunk 2]
```diff
@@ -821,6 +821,14 @@ def split_command(command):
     return shlex.split(command)
 
 
+def format_environment(environment):
+    def format_env(key, value):
+        if not value:
```
Comment (dnephin) [inline docker/utils/utils.py]: hmm, I just looked at this again, and maybe this needs to be `if value is None`

Comment (shin-) [inline docker/utils/utils.py]: ah, yeah, good point.

**Pull request merged**